easybuild/easyblocks/l/lammps.py

```diff
@@ -228,11 +228,91 @@ def update_kokkos_cpu_mapping(self):
         if LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('2Aug2023')):
             self.kokkos_cpu_mapping['icelake'] = 'ICX'
             self.kokkos_cpu_mapping['sapphirerapids'] = 'SPR'
+            self.kokkos_cpu_mapping['skylake'] = 'SKL'
         if LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('4Feb2025')):
             self.kokkos_cpu_mapping['neoverse_v2'] = 'ARMv9-Grace'
         if LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('2Apr2025')):
             self.kokkos_cpu_mapping['zen4'] = 'ZEN4'
 
+    def get_kokkos_arch(self, cuda_cc, kokkos_arch):
+        """
+        Return KOKKOS ARCH in LAMMPS required format, which is 'CPU_ARCH' and 'GPU_ARCH'.
+
+        see: https://docs.lammps.org/Build_extras.html#kokkos
+        """
+        # CPU arch
+        # NOTE: if the CPU KOKKOS_ARCH flag is specified, Kokkos will add the correspondent `-march` and `-mtune` flags
+        # to the compiler flags, which may override the ones set by EasyBuild.
+        # https://github.com/lammps/lammps/blob/stable_29Aug2024/lib/kokkos/cmake/kokkos_arch.cmake#L228-L531
+        processor_arch = None
+        if build_option('optarch') == OPTARCH_GENERIC:
+            # For generic Arm builds we use an existing target;
+            # this ensures that KOKKOS_ARCH_ARM_NEON is enabled (Neon is required for armv8-a).
+            # For other architectures we set a custom/non-existent type, which will disable all optimizations,
+            # and it should use the compiler (optimization) flags set by EasyBuild for this architecture.
+            if get_cpu_architecture() == AARCH64:
+                processor_arch = 'ARMV80'
+            else:
+                processor_arch = 'EASYBUILD_GENERIC'
+
+            _log.info("Generic build requested, setting CPU ARCH to %s." % processor_arch)
+            if kokkos_arch:
+                msg = "The specified kokkos_arch (%s) will be ignored " % kokkos_arch
+                msg += "because a generic build was requested (via --optarch=GENERIC)"
+                print_warning(msg)
+        elif kokkos_arch:
+            if kokkos_arch not in KOKKOS_CPU_ARCH_LIST:
+                warning_msg = "Specified CPU ARCH (%s) " % kokkos_arch
+                warning_msg += "was not found in listed options [%s]." % KOKKOS_CPU_ARCH_LIST
+                warning_msg += "Still might work though."
+                print_warning(warning_msg)
+            processor_arch = kokkos_arch
+
+        # If kokkos_arch was not set...
+        elif LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('2Aug2023')):
```

Comment:
```suggestion
        elif LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('22JUL2025')):
```
Since we have `archspec` as  a dependency in all the exhisting easyconfigs I prefer to start doing this from the next stable release.